app/anti_virus_check.py

```diff
@@ -0,0 +1,136 @@
+import collections
+import time
+
+import requests
+from sdc.rabbit.exceptions import QuarantinableError, RetryableError, BadMessageError
+
+from app import create_and_wrap_logger
+from app import settings
+
+logger = create_and_wrap_logger(__name__)
+
+AVResult = collections.namedtuple('AVResult', 'safe ready scan_results')
+
+
+class AntiVirusCheck:
+    def __init__(self, tx_id):
+        self.bound_logger = logger.bind(tx_id=tx_id)
+
+    def send_for_av_scan(self, payload):
+        self.bound_logger.info("Sending for AV check", filename=payload.file_name)
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        self.bound_logger.info("Sent for A/V check", data_id=data_id)
+
+        attempts = 0
+        while attempts <= settings.ANTI_VIRUS_MAX_ATTEMPTS:
+            attempts += 1
+            results = self._get_anti_virus_result(data_id)
+            if not results.ready:
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
+            elif not results.safe:
+                error = "Unsafe file detected for case id {} with filename {}".format(payload.case_id,
+                                                                                      payload.file_name)
+                self._write_scan_report(results, payload.file_name)
+                self.bound_logger.error(error)
```

Comment:
I don't think this is an error level event as nothing has actually gone wrong with the service.

Reply:
I see your point, i'll change to warning. Though we'll need to set up a specific alert in splunk for this then